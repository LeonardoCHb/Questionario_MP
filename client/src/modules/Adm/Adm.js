/* eslint-disable prettier/prettier */
import { Paper, Container } from "@material-ui/core";
import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import { getUsers, getUsersResponses, findAdm } from "../../actions/adm";
import { quizAllQuestions } from "../../actions/quiz";
import Table from "./components/Table/Table";
import styles from "./styles.js";

const useStyles = styles;

const Adm = () => {
  const dispatch = useDispatch();
  const users = useSelector((state) => state.getUsers);
  const usersResponses = useSelector((state) => state.getUsersResponses);
  const userQuizzes = useSelector((state) => state.quizAllQuestions);
  const userInfo = JSON.parse(localStorage.getItem("profile"));
  const isAdm = useSelector((state) => state.findAdm);
  const classes = useStyles();
  const [newUsers, setNewUsers] = useState([]);

  console.log(userInfo?.result._id);
  console.log(isAdm.id)

  const responses = (id) => {
    let filtered = [];
    if (userQuizzes) {
      filtered = usersResponses.filter((userRes) => {
        return userRes.answeredBy === id;
      });
    }
    return filtered;
  };
  // fetch all quizzes
  const quizzes = (id) => {
    let filtered = [];
    if (userQuizzes) {
      filtered = userQuizzes.filter((userQuiz) => {
        return userQuiz.creator === id;
      });
    }
    return filtered;
  };

  useEffect(() => {
    let userTableData = [];
    if (users) {
      userTableData = users.map((user) => ({
        id: user._id,
        email: user.email,
        name: user.name,
        QtdQuizzes: quizzes(user._id).length,
        QtdResponses: responses(user._id).length,
        quizResponses: responses(user._id),
        quizCreated: quizzes(user._id),
      }));
      setNewUsers(userTableData);
    }
  }, [users, usersResponses, userQuizzes]);

  useEffect(() => {
    dispatch(quizAllQuestions());
    dispatch(getUsers());
    dispatch(getUsersResponses());
    dispatch(findAdm());
  }, []);

  return isAdm.id === userInfo?.result._id && isAdm.id !== undefined ? (
    <form>
      <Container
        className={classes.TableContainer}
        component="main"
        maxWidth="md"
      >
        <Paper>
          <Table autoPageSize="true" users={newUsers} />
        </Paper>
      </Container>
    </form>
  ) : (
    <React.Fragment>
     <h1>==============VOCÊ É O IMPOSTOR========================================</h1>
      <h1>
⠀⠀⠀⠀⠀⠀⠀⠀⢀⣴⣿⡿⠛⠉⠙⠛⠛⠛⠛⠻⢿⣿⣷⣤⡀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠀⣼⣿⠋⠀⠀⠀⠀⠀⠀⠀⢀⣀⣀⠈⢻⣿⣿⡄
⠀⠀⠀⠀⠀⠀⠀⣸⣿⡏⠀⠀⠀⣠⣶⣾⣿⣿⣿⠿⠿⠿⢿⣿⣿⣿⣄⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⣿⣿⠁⠀⠀⢰⣿⣿⣯⠁⠀⠀⠀⠀⠀⠀⠀⠈⠙⢿⣷⡄
⠀⠀⣀⣤⣴⣶⣶⣿⡟⠀⠀⠀⢸⣿⣿⣿⣆⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣿⣷⠀
⠀⢰⣿⡟⠋⠉⣹⣿⡇⠀⠀⠀⠘⣿⣿⣿⣿⣷⣦⣤⣤⣤⣶⣶⣶⣶⣿⣿⣿
⠀⢸⣿⡇⠀⠀⣿⣿⡇⠀⠀⠀⠀⠹⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡿⠃
⠀⣸⣿⡇⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠉⠻⠿⣿⣿⣿⣿⡿⠿⠿⠛⢻⣿⡇
⠀⣿⣿⠁⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⣧
⠀⣿⣿⠀⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⣿
⠀⣿⣿⠀⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⣿
⠀⢿⣿⡆⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⡇⠀⠀
⠀⠸⣿⣧⡀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣿⣿⠃⠀
⠀⠀⠛⢿⣿⣿⣿⣿⣇⠀⠀⠀⠀⠀⣰⣿⣿⣷⣶⣶⣶⣶⠶⠀⢠⣿⣿⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⣿⣿⠀⠀⠀⠀⠀⣿⣿⡇⠀⣽⣿⡏⠁⠀⠀⢸⣿⡇⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⣿⣿⠀⠀⠀⠀⠀⣿⣿⡇⠀⢹⣿⡆⠀⠀⠀⣸⣿⠇⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⢿⣿⣦⣄⣀⣠⣴⣿⣿⠁⠀⠈⠻⣿⣿⣿⣿⡿⠏⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠀⠈⠛⠻⠿⠿⠿⠿⠋⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
      </h1>
    </React.Fragment>
  );
};

export default {
  routeProps: {
    path: "/adm",
    component: Adm,
  },
  name: "Adm",
};
