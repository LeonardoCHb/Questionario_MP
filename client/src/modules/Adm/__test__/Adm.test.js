import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { mount, configure } from "enzyme";
import EnzymeToJson from "enzyme-to-json";
import React from "react";
import { Provider } from "react-redux";
import * as reactRedux from "react-redux";
import { ToastProvider } from "react-toast-notifications";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

import Adm from "../Adm";

const AdmComponent = Adm.routeProps.component;

const user = {
  createdAt: "2021-04-30T23:54:53.614Z",
  _id: "608cce84d339170cd8ec9894",
  name: "Fernando",
  email: "fernando@email",
  password: "$2a$12$CDydrU0e8OwZZZNg0pWrgeuRBHaDimQplLKxBwNmDPvlrDJ24J69y",
  __v: 0,
};

localStorage.setItem("profile", JSON.stringify({ result: { ...user } }));

const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");

beforeEach(() => {
  useSelectorMock.mockClear();
  useDispatchMock.mockClear();
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const initialState = [];
const store = mockStore(initialState);

configure({ adapter: new Adapter() });
it("Adm Ã© renderizado ", () => {
  useSelectorMock.mockReturnValue([{ ...user }]);

  const subject = mount(
    <Provider store={store}>
      <ToastProvider placement="bottom-right">
        <AdmComponent />
      </ToastProvider>
    </Provider>
  );
  expect(EnzymeToJson(subject)).toMatchSnapshot();
});
