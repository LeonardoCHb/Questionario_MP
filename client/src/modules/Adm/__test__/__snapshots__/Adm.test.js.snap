// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adm é renderizado  1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ToastProvider
    autoDismiss={false}
    autoDismissTimeout={5000}
    components={
      Object {
        "Toast": [Function],
        "ToastContainer": [Function],
      }
    }
    placement="bottom-right"
    transitionDuration={220}
  >
    <Adm>
      <h1>
        ==============VOCÊ É O IMPOSTOR========================================
      </h1>
      <h1>
        ⠀⠀⠀⠀⠀⠀⠀⠀⢀⣴⣿⡿⠛⠉⠙⠛⠛⠛⠛⠻⢿⣿⣷⣤⡀⠀⠀⠀⠀⠀ ⠀⠀⠀⠀⠀⠀⠀⠀⣼⣿⠋⠀⠀⠀⠀⠀⠀⠀⢀⣀⣀⠈⢻⣿⣿⡄ ⠀⠀⠀⠀⠀⠀⠀⣸⣿⡏⠀⠀⠀⣠⣶⣾⣿⣿⣿⠿⠿⠿⢿⣿⣿⣿⣄⠀⠀⠀ ⠀⠀⠀⠀⠀⠀⠀⣿⣿⠁⠀⠀⢰⣿⣿⣯⠁⠀⠀⠀⠀⠀⠀⠀⠈⠙⢿⣷⡄ ⠀⠀⣀⣤⣴⣶⣶⣿⡟⠀⠀⠀⢸⣿⣿⣿⣆⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣿⣷⠀ ⠀⢰⣿⡟⠋⠉⣹⣿⡇⠀⠀⠀⠘⣿⣿⣿⣿⣷⣦⣤⣤⣤⣶⣶⣶⣶⣿⣿⣿ ⠀⢸⣿⡇⠀⠀⣿⣿⡇⠀⠀⠀⠀⠹⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡿⠃ ⠀⣸⣿⡇⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠉⠻⠿⣿⣿⣿⣿⡿⠿⠿⠛⢻⣿⡇ ⠀⣿⣿⠁⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⣧ ⠀⣿⣿⠀⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⣿ ⠀⣿⣿⠀⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⣿ ⠀⢿⣿⡆⠀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⢸⣿⡇⠀⠀ ⠀⠸⣿⣧⡀⠀⣿⣿⡇⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⣿⣿⠃⠀ ⠀⠀⠛⢿⣿⣿⣿⣿⣇⠀⠀⠀⠀⠀⣰⣿⣿⣷⣶⣶⣶⣶⠶⠀⢠⣿⣿⠀⠀⠀ ⠀⠀⠀⠀⠀⠀⠀⣿⣿⠀⠀⠀⠀⠀⣿⣿⡇⠀⣽⣿⡏⠁⠀⠀⢸⣿⡇⠀⠀⠀ ⠀⠀⠀⠀⠀⠀⠀⣿⣿⠀⠀⠀⠀⠀⣿⣿⡇⠀⢹⣿⡆⠀⠀⠀⣸⣿⠇⠀⠀⠀ ⠀⠀⠀⠀⠀⠀⠀⢿⣿⣦⣄⣀⣠⣴⣿⣿⠁⠀⠈⠻⣿⣿⣿⣿⡿⠏⠀⠀⠀⠀ ⠀⠀⠀⠀⠀⠀⠀⠈⠛⠻⠿⠿⠿⠿⠋⠁⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀
      </h1>
    </Adm>
    <Portal
      containerInfo={
        <body>
          <div
            class="react-toast-notifications__container css-omekjv-ToastContainer"
          />
        </body>
      }
    >
      <ToastContainer
        hasToasts={false}
        placement="bottom-right"
      >
        <EmotionCssPropInternal
          __EMOTION_LABEL_PLEASE_DO_NOT_USE__="ToastContainer"
          __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
          className="react-toast-notifications__container"
          css={
            Object {
              "bottom": 0,
              "boxSizing": "border-box",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "overflow": "hidden",
              "padding": 8,
              "pointerEvents": "none",
              "position": "fixed",
              "right": 0,
              "zIndex": 1000,
            }
          }
        >
          <div
            className="react-toast-notifications__container css-omekjv-ToastContainer"
          >
            <TransitionGroup
              childFactory={[Function]}
              component={null}
            />
          </div>
        </EmotionCssPropInternal>
      </ToastContainer>
    </Portal>
  </ToastProvider>
</Provider>
`;
